File: 1-iniciante/1040.py
grades = input().split()

average = ((float(grades[0]) * 2.0) + (float(grades[1]) * 3.0) + (float(grades[2]) * 4.0) + (float(grades[3]) * 1.0)) / (2.0 + 3.0 + 4.0 + 1.0)

print('Media: %.1f' %(average))

if average >= 7.0:
  print('Aluno aprovado.')

elif average < 5.0:
  print('Aluno reprovado.')

else:
  print('Aluno em exame.')

  # exam grade
  grades[0] = float(input())
  print('Nota do exame: %.1f' %(grades[0]))

  grades[0] = (grades[0] + average) / 2.0

  if grades[0] >= 5.0:
    print('Aluno aprovado.')

  else:
    print('Aluno reprovado.')

  print('Media final: %.1f' %(grades[0]));

File: 1-iniciante/1040.c
#include <stdio.h>

int main() {
  double grades[4]={0.0}, average=0.0;

  scanf("%lf%lf%lf%lf", &grades[0], &grades[1], &grades[2], &grades[3]);

  average = ((grades[0] * 2.0) + (grades[1] * 3.0) + (grades[2] * 4.0) + (grades[3] * 1.0)) / (2.0 + 3.0 + 4.0 + 1.0);

  printf("Media: %.1lf\n", average);

  if(average >= 7.0) {
    printf("Aluno aprovado.\n");

  }else if(average < 5.0) {
    printf("Aluno reprovado.\n");

  }else {
    printf("Aluno em exame.\n");

    // exam grade
    scanf("%lf", &grades[0]);
    printf("Nota do exame: %.1lf\n", grades[0]);

    grades[0] = (grades[0] + average) / 2.0;

    if(grades[0] >= 5.0) {
      printf("Aluno aprovado.\n");

    }else {
      printf("Aluno reprovado.\n");

    }

    printf("Media final: %.1lf\n", grades[0]);
  }

  return 0;
}